tests/test_media_handlers.py

```diff
@@ -262,3 +263,25 @@ async def on_post(self, req, resp):
     result = testing.simulate_post(app, '/', json=doc)
     assert result.status_code == 200
     assert result.json == [None]
+
+
+def test_json_err_no_handler(asgi):
+    app = create_app(asgi)
+
+    handlers = media.Handlers({falcon.MEDIA_URLENCODED: media.URLEncodedFormHandler()})
+    app.req_options.media_handlers = handlers
+    app.resp_options.media_handlers = handlers
+
+    class Resource:
+        def on_get(self, req, resp):
+            raise falcon.HTTPForbidden()
+
+    class ResourceAsync:
+        async def on_get(self, req, resp):
+            raise falcon.HTTPForbidden()
+
+    app.add_route('/', ResourceAsync() if asgi else Resource())
```

Comment:
We would automatically wrap sync resources for async routes, as explained on another PR.
So we could DRY this a bit.

Reply:
so this would work also without the async resource?

Reply:
i guess that this is what's the env variable `FALCON_ASGI_WRAP_NON_COROUTINES` is for